File: two_pointers/two_sum_2/main.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l, r = 0, len(numbers) - 1

        while l < r:
            if numbers[l] + numbers[r] == target:
                return [min(l, r) + 1, max(l, r) + 1]
            
            if numbers[l] + numbers[r] > target:
                r -= 1
            else:
                l += 1

File: two_pointers/two_sum_2/main.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& numbers, int target) {
        int l = 0;
        int r = numbers.size() - 1;

        while (l < r) {
            if (numbers[l] + numbers[r] == target) {
                return (vector<int> {min(l, r) + 1, max(l, r) + 1});
            }

            if (numbers[l] + numbers[r] > target) {
                r--;
            } else{
                l++;
            }
        }
    }
};
